std/traits.d

```diff
@@ -2513,11 +2513,8 @@ Returns:
 template isInnerClass(T)
 if (is(T == class))
 {
-    import std.meta : staticIndexOf;
-
     static if (is(typeof(T.outer)))
-        enum isInnerClass = __traits(isSame, typeof(T.outer), __traits(parent, T))
-                         && (staticIndexOf!(__traits(allMembers, T), "outer") == -1);
+        enum isInnerClass = __traits(isSame, typeof(T.outer), __traits(parent, T));
```

Comment:
@TurkeyMan this implementation is really simple and uses built-in features, would this be the better of the two?

Reply:
Ummm, this test is incomplete (as discussed in the first comment thread above). This test raises false positives when a type has an 'outer' member, which we need to negate with the search for an outer member. A real `__traits` would be better, but in lieu of that, it probably should still work properly.

Reply:
I'm also not sure what you mean by 'built in features'? My test is identical, except with an extra clause that catches the false-positive case.